gcn_classic_text_to_json/notices/calet/__init__.py

```diff
@@ -0,0 +1,111 @@
+        "foreground_duration": ("FOREGND_DUR", "float"),
+        "background_duration": ("BACKGND_DUR1", "float"),
+    },
+}
+
+
+def text_to_json_calet(notice, record_number):
+    """Function calls text_to_json and then adds additional fields with cannot be dealt with by the general function.
+
+    Parameters
+    -----------
+    notice: dict
+        The text notice that is being parsed.
+    record_number: int
+        The current notice in the webpage being parsed.
+    notice_start_idx: int
+        The index at which the current text notice begins.
+
+    Returns
+    -------
+    dictionary
+        A dictionary compliant with the associated schema for the mission."""
+    output_dict = conversion.text_to_json(notice, input)
+
+    output_dict["record_number"] = record_number
+    if record_number == 1:
+        output_dict["alert_type"] = "initial"
+    else:
+        output_dict["alert_type"] = "update"
+
+    output_dict["mission"] = "CALET"
+    output_dict["instrument"] = "GBM"
+    output_dict["trigger_type"] = "rate"
+
+    lon_lat_data = notice["SC_LON_LAT"].split(",")
+    lon, lat = lon_lat_data[0], lon_lat_data[1][:5]
+    output_dict["longitude"] = float(lon)
+    output_dict["latitude"] = float(lat)
+
+    energy_data = notice["ENERGY_BAND"]
+    energy_low, energy_high = energy_data.split()[0].split("-")
+    output_dict["rate_energy_range"] = [int(energy_low), int(energy_high)]
+
+    detector_data = notice["TRIGGER_DET"].split()
+    detector_opts = ["on", "triggered"]
+    output_dict["detector_status"] = {
+        detector_data[-3][1:]: detector_opts[int(detector_data[0])],
+        detector_data[-2]: detector_opts[int(detector_data[1])],
+        detector_data[-1][:-1]: detector_opts[int(detector_data[2])],
+    }
+
+    return output_dict
+
+
+def create_all_calet_jsons():
+    """Creates a `calet_json` directory and fills it with the json for all CALET triggers."""
+    output_path = "./calet_jsons/"
+    if not os.path.exists(output_path):
+        os.makedirs(output_path)
+
+    archive_link = "https://gcn.gsfc.nasa.gov/calet_triggers.html"
+    prefix = "https://gcn.gsfc.nasa.gov/"
+    links_list = conversion.parse_trigger_links(archive_link, prefix)
+
+    for link in links_list:
+        data = requests.get(link).text
+        trig_num = utils.find_string_between(link, "other/", ".calet")
+
+        record_number = 1
+        start_idx = data.find("\n") + 1
+        while True:
+            end_idx = data.find("\n \n ", start_idx)
+            notice_message = email.message_from_string(data[start_idx:end_idx].strip())
+            comment = "".join(notice_message.get_all("COMMENTS"))
+            notice_dict = dict(notice_message)
+            notice_dict["COMMENTS"] = comment
+
+            output = text_to_json_calet(notice_dict, record_number)
+
+            f = open(f"{output_path}CALET_GBM_{trig_num}_{record_number}.json", "w")
+            json.dump(output, f)
+            f.close()
```

Comment:
```suggestion
            with open(f"{output_path}CALET_GBM_{trig_num}_{record_number}.json", "w") as f:
                json.dump(output, f)
```

Reply:
Sorry, I should have included this in the same suggested change I made previously